tests/test_config_validation.py

```
@@ -69,6 +70,25 @@ def test_valid_backend_accepted(self):
+        config.backend = "fast-pysf"
+        monkeypatch.setattr(config_validation, "list_backends", lambda: ["dummy"])
+
+        _check_backend_valid(config)
+
+    def test_non_optional_missing_backend_still_raises(self, monkeypatch: pytest.MonkeyPatch):
+        """Only the canonical optional backend may bypass registry validation."""
+        config = RobotSimulationConfig()
+        config.backend = "nonexistent_backend"
+        monkeypatch.setattr(config_validation, "list_backends", lambda: ["dummy"])
+
+        with pytest.raises(KeyError, match="Unknown backend.*Available backends"):
```

Ruff (0.15.17)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)